modules/ROOT/pages/core-operations.adoc

```diff
@@ -239,12 +236,14 @@ xref:concurrent-mutations-cluster.adoc[CAS] values are not used with counter ope
 The intent of the counter operation is to simply increment the current server-side value of the document.
 If you wish to only increment the document if it is at a certain value, then you may use a normal [.api]`upsert` function with CAS:
 
-[source,python]
+[source,java]
 ----
-rv = cb.get('counter_id')
-value, cas = rv.value, rv.cas
-if should_increment_value(value):
-  cb.upsert('counter_id', value + increment_amount, cas=cas)
+JsonDocument counter = bucket.get("counter_id");
+long cas = document.cas();
+long value = document.content();
```

Comment:
Can you please check these examples compile.  None of this looks correct: upsert doesn't take an execute, you get document with variable 'counter' but use 'document' instead, 'document.content()' returns a JsonObject not a long.

Reply:
Yes. Will correct this.